### PR TITLE
Added cmdlet Test-VSTeamYamlPipelines for testing YAML pipeline code

### PR DESCRIPTION
**Pull request by SebastianSchuetze**
# PR Summary

added a new cmdlet for a new API that was released the last sprint.

https://docs.microsoft.com/en-us/azure/devops/release-notes/2020/sprint-165-update#azure-pipelines-1

I still need to add some documentation first and change the changelog.

## PR Checklist

- [x] [Write Help](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-help)
- [x] [Write Unit Test](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-unit-test)
- [x] [Update CHANGELOG.md](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#add-a-format-file)

### CODE REVIEW

File: Source/Public/Test-VSTeamYamlPipeline.ps1 [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+         $body.YamlOverride = [string](Get-Content -raw $FilePath)
+      }
+      elseif ($YamlOverride) {
+         $body.YamlOverride = $YamlOverride         
+      }
+
+      try {
+         # Call the REST API
+         $resp = _callAPI -ProjectName $ProjectName -Area 'pipelines' -Resource "$PipelineId" -id "runs" `
+            -Method Post -ContentType 'application/json' -Body ($body | ConvertTo-Json) `
```
Comment (MichelZ) [inline Source/Public/Test-VSTeamYamlPipeline.ps1]: ; charset=utf-8
:)

Comment (SebastianSchuetze) [reply Source/Public/Test-VSTeamYamlPipeline.ps1]: we need to make that maybe part of the build pipeline probably of the API call to not allow non utf json calls.

**Comment by tomkerkhove** [comment]
Do we need 'VSTeam' prefix? For consistency reasons?

File: .docs/Test-VSTeamYamlPipeline.md [hunk 2]
```diff
@@ -19,13 +19,31 @@
...
+PS C:\> $yamlOverride = [string](Get-Content -raw $FilePath)
+PS C:\> Test-VSTeamYamlPipeline -Project DemoProject -PipelineId 24 -YamlOverride $yamlOverride
+```
+
+This example checks the YAML pipeline with ID 24 and the content of a yaml file in the variable $yamlOverride for consistency on Azure DevOps to see if the changes still work.
+
+### -------------------------- EXAMPLE 3 --------------------------
+
+```PowerShell
+PS C:\> $yamlOverride = [string](Get-Content -raw $FilePath)
```
[the remark this answers is not in the document]
Comment (SebastianSchuetze) [reply .docs/Test-VSTeamYamlPipeline.md]: I added the option to either provide the file content or the file itself. This is what it makes clearer for me. Instead, I didn't want to provide an example YAML

**Comment by SebastianSchuetze** [comment]
> Do we need 'VSTeam' prefix? For consistency reasons?

The whole module is called VSTeam and every cmdlet has it as well. Which is consistent with the e.g. AzureAD, Az, AzureRM modules. This is how you prevent also conflicts with other modules the are loaded as well. 
See it a namespace.

**Comment by SebastianSchuetze** [comment]
@JustinGrote do you see any useful parameter that might make sense to add?

**Comment by tomkerkhove** [comment]
Sounds good to me!

**Pull request merged**